tokio/src/runtime/time/mod.rs

```diff
@@ -470,5 +494,31 @@ impl fmt::Debug for Inner {
     }
 }
 
+// ===== impl ShardedWheel =====
+
+impl ShardedWheel {
+    /// Locks the driver's sharded wheel structure.
+    pub(super) fn lock_sharded_wheel(
+        &self,
+        shard_id: u32,
+    ) -> crate::loom::sync::MutexGuard<'_, Wheel> {
+        let index = shard_id % (self.0.len() as u32);
+        // Safety: This modulo operation ensures that the index is not out of bounds.
+        unsafe { self.0.get_unchecked(index as usize) }.lock()
+    }
+
+    /// Gets a mutable reference to the sharded wheel with the given id.
+    pub(super) fn get_sharded_wheel(&mut self, shard_id: u32) -> &mut wheel::Wheel {
```

Comment:
Perhaps this ought to be `get_sharded_wheel_mut`? Not a big deal either way though.